crates/storage/nippy-jar/src/writer.rs

```diff
@@ -1,15 +1,42 @@
 use crate::{compression::Compression, ColumnResult, NippyJar, NippyJarError, NippyJarHeader};
 use std::{
     cmp::Ordering,
-    fmt,
     fs::{File, OpenOptions},
     io::{Read, Seek, SeekFrom, Write},
+    ops::{Deref, DerefMut},
     path::Path,
 };
 
 /// Size of one offset in bytes.
 const OFFSET_SIZE_BYTES: u64 = 8;
 
+/// Holds a reference or an owned [`NippyJar`].
+#[derive(Debug)]
+enum JarHolder<'a, H: NippyJarHeader = ()> {
```

Comment:
nit: `JarCow`/`CowJar` makes more sense I think? since this is effectively a `Cow` (except its `&mut`)